app/registries/foreman/setting_manager.rb

```diff
@@ -0,0 +1,72 @@
+module Foreman
+  class SettingManager
+    class << self
+      # Holds all the setting definitions for application
+      def settings
+        @settings ||= {}
+      end
+
+      # Holds all setting categories with their labels
+      def categories
+        @categories ||= { 'general' => N_('General') }
+      end
+
+      def define(context_name, &block)
+        SettingMapper.new(self, context_name).instance_eval(&block)
+      end
+    end
+
+    class SettingMapper
+      attr_reader :context_name
+
+      def initialize(manager, context_name)
+        @manager = manager
+        @context_name = context_name
+      end
+
+      def category(category_name, category_label = nil, &block)
+        @manager.categories[category_name.to_s] ||= category_label
+        CategoryMapper.new(@manager.settings, context_name, category_name.to_s).instance_eval(&block)
+      end
+    end
```

Comment:
Can't this be merged into a single class? `SettingManager` only has static methods while `SettingMapper` only has instance methods.

Reply:
It can, it is virtually the same thing, I just like the responsibilities being quite strictly divided, but I don't care much.
I'll merge into one :+1:

Reply:
Separation can be a benefit, I was mostly wondering. Perhaps I need to rethink it after some sleep.

Reply:
Sure :) I can always go back it's simple change, I don't mind eighter way, leaning a bit towards a separate class, let me know when you think about it :)